### PR TITLE
BUG FIX: Raise appropriate strings error when concatenating strings column

### PR DESCRIPTION
**Pull request by skirui-source**
Fixes: https://github.com/rapidsai/cudf/issues/8228

### CODE REVIEW

File: python/cudf/cudf/core/column/column.py [hunk 1]
```diff
@@ -282,7 +282,15 @@ def _concat(
 
         # Perform the actual concatenation
         if newsize > 0:
-            col = libcudf.concat.concat_columns(objs)
+            try:
+                col = libcudf.concat.concat_columns(objs)
+            except RuntimeError as e:
+                if "concatenated rows exceeds size_type range" in str(e):
```
Comment (davidwendt) [inline python/cudf/cudf/core/column/column.py:288]: This error from libcudf is not specific to strings columns. The number of rows in any column cannot exceed `size_type`. This means concatenating 2 integer columns each with rows lengths greater than `size_type/2` would throw this error from libcudf.

The specific message for strings below only occurs for strings columns.

**Comment by JohnZed** [comment]
@skirui-source / @davidwendt / @quasiben - is this needed for 21.06 or should it be pushed back?

File: python/cudf/cudf/core/column/column.py (hunk 1 above)
Comment (skirui-source) [reply python/cudf/cudf/core/column/column.py]: I generalized the error to `"total size of output is too large for a cudf column"`, is that sufficient?

**Review by shwina — APPROVED** [review]
LGTM!

**Review by isVoid — APPROVED** [review]
👍

**Comment by shwina** [comment]
@gpucibot merge

**Pull request merged**